### DISCUSSION TITLE
How to distinguish auto exported metrics (e.g.  process.runtime.go.mem.heap_alloc) from different hosts?

### DISCUSSION THREAD

**Discussion by cs-ranbi** [open-telemetry/opentelemetry-go#5156] [Q&A]
I use OTLP to export metrics to Grafana.
I uas a "host" attribute to distinguish metrics with the same name on different hosts.
However, I noticed there are a couple of metrics get auto created by the sdk: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/runtime/runtime.go#L111-L131
Is it possible to distinguish these metrics from different hosts by adding additional attributes to them?

**Comment by dmathieu** [comment]
You can setup a view in your code to change the attributes/names of auto-generated metrics.
https://opentelemetry.io/docs/languages/go/instrumentation/#registering-views

**Reply by cs-ranbi** [reply]
Thanks for the response.
I think I am only able to filter attributes, but I don't see how to rename attributes from the API.

**Reply by dmathieu** [reply]
Actually, since what you need to do is add attributes and not modify the emitted metrics, what you're looking for is probably resources.
See https://opentelemetry.io/docs/languages/go/resources/

This example is about traces, but setting resources for a meter provider works exactly the same way.
See https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#WithResource
